server/src/main/java/org/candlepin/resource/util/GuestMigration.java

```diff
@@ -66,6 +69,34 @@ public void migrate(boolean flush) {
         migrationPending = false;
     }
 
+    /**
+     * Build a manifest detailing any guest migrations occurring due to a host consumer update.
+     *
+     * If a consumer's guest was already reported by another host consumer, record that host in the
+     * manifest as well so that the migration can be made atomically.
+     *
+     * @param incoming incoming consumer in DTO form
+     * @param existing existing consumer in model form
+     * @return guestMigration object that contains all information related to the migration
+     */
+    public GuestMigration buildMigrationManifest(ConsumerDTO incoming, Consumer existing) {
+        if (incoming.getGuestIds() == null) {
+            log.debug("Guests not included in this consumer update, skipping update.");
+            migrationPending = false;
+            return this;
+        }
+
+        Consumer incomingConsumerEntity = new Consumer();
```

Comment:
There is almost as much work done here just to convert the consumer as there would be if the DTO were used directly. This is the type of thing we want to avoid.

Reply:
@Ceiu The original version of this method is called from quite a few other places using a Consumer (as opposed to ConsumerDTO) for a parameter, so both versions of a parameter need to be available. And this whole class, and its nested MigrationManifest class, are very coupled to Consumer and GuestId. The only alternative to this wrapper that translates to a Consumer would be to write a whole new class called something like 'GuestMigrationDTO' that is using ConsumerDTO and GuestIdDTO alongside it. Would that be an acceptable solution?